docs/nebula-design-concepts.md

```diff
@@ -0,0 +1,214 @@
+
+In the preceding example, the schema of **player** has three kind of properties:
+
+- id (vid)
+- Name (string)
+- Age (int)
+
+ the schema of **serve** has two kinds of properties:
+
+ - start_year (int)
+ - end_year (int)
+
+
+### Edgetype
+
+Each edge has an edgetype, and each edgetype defines the properties of the edge.
+
+This part is written based on my own understand.
+-->
+
+
+### Vertex, type of vertex (tag) and the corresponding vertex properties
+
+In Nebula Graph, the type of vertex is called **tag**. **A vertex must be labeled with at least one tag. Each tag may have a set of properties called schema.**
+
+Consider the above picture as example, there are two tags: **player** and **team**. Player has three properties: id (vid), Name(string), and Age(int); team has two properties: id (vid) and Name(string).
+
+Like Mysql, Nebula Graph is a strong schema database. The property name and data type are determined before the data is written.
+
+### Edge, edgetype and the corresponding edge properties
+
+**Edges in Nebula Graph always have a type, a start (src)and an end vertex (dst), and a direction.** They can be self-referencing/looping but can never be dangling (missing a start or end node). **Each edge has an edgetype, and each edgetype defines the properties of the edge.**
+
+In the preceding example, there are two kinds of edges, one is **like** pointing
+from player to player, the property is likeness (double); the other is **serve**,
+pointing from team to serve, the properties are start_year(int) and end_year(int).
+
+It should be noted that there may be **multiple edges of the same or different types
+at the same time** between the start (src)and end vertex (dst).
+
+
+
+## Graph partition
+
+![image](https://user-images.githubusercontent.com/42762957/61119934-76829b80-a4ce-11e9-9d49-2abb11b5f7b2.png)
+
+
+In applications such as social networks, or road networks, the number of vertices and their connections are extremely large. Therefore, partitioning large graphs is inevitable for complexity reduction and parallelization. Nebula Graph adopts **Edge Separators** to cut a graph into smaller components. And the partition number cannot be changed once being configured.
+
+## Data model
+
+
+![image](https://user-images.githubusercontent.com/42762957/61120073-b0ec3880-a4ce-11e9-975f-c19482d4b109.png)
+
+
+
+Data is stored as key-value pairs in Nebula Graph. For vertices, as illustrated in the diagram above, the key consists of four parts: partID (4 bit), vertexID (8 bit), tagID (4 bit) and Version (8 bit). And they are spread across various partitions based on the hash of their vertexIds (vid).
+
+For edges, the key consists of two separate key-values, i.e. out-key and
+in-key, respectively. The out-key and the edge's corresponding starting point are stored in one partition, while the in-key and end point are stored in another one.
+
```

Comment:
For edges,  the out-edges are stored together with source point,  while the in-edges are together with destination point.   The difference for out-edge key and in-edge key is the edgeType,  which is negative for in-edge key.